### ISSUE TITLE
Common: pom produced doesn't declare dependencies and should

### ISSUE THREAD

**Issue by michaelayates** [snowplow/snowplow-android-tracker#114]
The POM produced as part of the gradle release lives here:
http://maven.snplow.com/releases/com/snowplowanalytics/snowplow-android-tracker/0.3.0/snowplow-android-tracker-0.3.0.pom

It looks like this:
<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
 <modelVersion>4.0.0</modelVersion>
 <groupId>com.snowplowanalytics</groupId>
  <artifactId>snowplow-android-tracker</artifactId>
  <version>0.3.0</version>
</project>

e.g. it has _no_ dependencies. This means when you include snowplow-android-tracker you manually have to add dependencies by inspecting the gradle build file per "For the full list of dependencies, please see our Gradle build file". This is messy.

The pom produced by gradle should add those as dependencies and let the dependency management of gradle/maven work out the full dependency set for a project.

According to the gradle doco here:
http://gradle.org/docs/current/userguide/maven_plugin.html
"Gradle can generate this POM for you using the dependency information it has"

At this stage this doesn't seem to be happening for the POM currently being built.

**Comment by jonalmeida** [comment]
Hey @michaelayates , this was a problem that I had faced with android-gradle a while back as well. The solution at the time was to add manually add the dependencies into the build.gradle, but a better solution that was proposed in [this SO question](http://stackoverflow.com/questions/25585537/importing-my-android-library-in-an-app-isnt-recognized-as-a-library) was to package the dependencies along with the tracker library itself.

The TL;DR is that android-gradle acts differently from the normal gradle so that doc is a bit misleading in this case.

**Comment by alexanderdean** [comment]
As per @jonalmeida's comment, I don't think we've yet found a way to do this within the confines of android-gradle's capabilities. Assigning to @jbeemster to follow up on.

**Comment by jbeemster** [comment]
Bumping this as have to learn a bit more about gradle still.

**Comment by jbeemster** [comment]
Hey @alexanderdean so with this ticket I am not sure how to advance with it.  The answer in this question (http://stackoverflow.com/questions/25585537/importing-my-android-library-in-an-app-isnt-recognized-as-a-library) pretty well states that defining dependencies in the POM for Android is a no go?

We also cannot package a JAR with all of our dependencies as we want to give people the option to include or not include certain libraries like google play services.

**Comment by alexanderdean** [comment]
Thanks @jbeemster - that's a helpful summary. I'll refer this back to @michaelayates - if you have any suggestions Michael let us know, other we will close this.

**Comment by michaelayates** [comment]
maven allows you to declare depdencies as optional. So you could add your true dependencies as non optional and if there are feature based optionalities declare them as optional?

https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html

**Comment by alexanderdean** [comment]
Does that work with android-gradle @michaelayates ?

**Comment by AlexBenny** [comment]
No issues have been reported since 2015. The POM file now looks different from what indicated as the origin of the issue.
